Tests/MoSoAnalyticsTests/GleanBaseTrackerTests.swift

```diff
@@ -0,0 +1,140 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Glean
+import MoSoCore
+import XCTest
+@testable import MoSoAnalytics
+
+final class GleanBaseTrackerTests: XCTestCase {
+    override func setUp() {
+        Glean.shared.resetGlean(clearStores: true)
```

Comment:
curious: why are we using the same singleton as in our app as opposed to using a mock or test object?

Reply:
@Gio2018 can answer better, but I think this is equivalent to our "snowplow micro" testing. Glean introduces some built-in testing support, so this test, while it is testing an implicit dependency, is ensuring that our events are correctly being sent to Glean, without having to set up an equivalent like Snowplow Micro (which I'm not sure even exists). From there, we have to assume that Glean is working as intended.

Reply:
ok thanks, this helps a lot!!

Reply:
Yep that's what It is, thanks @dskuza !!